tests/benchmarks/MicroPerf/Collections.fs

```diff
@@ -0,0 +1,298 @@
+        
+    [<Benchmark>]
+    member x.ListRemoveManyAtBeginning() =
+        async {
+            List.init x.Length (fun _ -> 0)
+            |> List.removeManyAt 0 10 
+            |> ignore
+        }
+        |> Async.StartAsTask
+
+    [<Benchmark>]
+    member x.ListRemoveManyAtEnd() =
+        async {
+            List.init x.Length (fun _ -> 0)
+            |> List.removeManyAt (x.Length - 11) 10 
+            |> ignore
+        }
+        |> Async.StartAsTask
+        
+    [<Benchmark>]
+    member x.ListInsertAtBeginning() =
+        async {
+            List.init x.Length (fun _ -> 0)
+            |> List.insertAt 0 1 
+            |> ignore
+        }
+        |> Async.StartAsTask
+
+    [<Benchmark>]
+    member x.ListInsertAtEnd() =
+        async {
+            List.init x.Length (fun _ -> 0)
+            |> List.insertAt (x.Length - 1) 1 
+            |> ignore
+        }
+        |> Async.StartAsTask
+        
+    [<Benchmark>]
+    member x.ListInsertManyAtBeginning() =
+        async {
+            List.init x.Length (fun _ -> 0)
+            |> List.insertManyAt 0 [1..10] 
+            |> ignore
+        }
+        |> Async.StartAsTask
+
+    [<Benchmark>]
+    member x.ListInsertManyAtEnd() =
+        async {
+            List.init x.Length (fun _ -> 0)
+            |> List.insertManyAt (x.Length - 11) [1..10]
+            |> ignore
+        }
+        |> Async.StartAsTask
+        
+    [<Benchmark>]
+    member x.ListUpdateAtBeginning() =
+        async {
+            List.init x.Length (fun _ -> 0)
+            |> List.updateAt 0 1 
+            |> ignore
+        }
+        |> Async.StartAsTask
+
+    [<Benchmark>]
+    member x.ListUpdateEnd() =
+        async {
+            List.init x.Length (fun _ -> 0)
+            |> List.updateAt (x.Length - 1) 1
+            |> ignore
+        }
+        |> Async.StartAsTask
+        
+    /// Array
+    [<Benchmark>]
+    member x.ArrayRemoveAtBeginning() =
+        async {
+            Array.init x.Length (fun _ -> 0)
+            |> Array.removeAt 0 
+            |> ignore
+        }
+        |> Async.StartAsTask
+
+    [<Benchmark>]
+    member x.ArrayRemoveAtEnd() =
+        async {
+            Array.init x.Length (fun _ -> 0)
+            |> Array.removeAt (x.Length - 1) 
+            |> ignore
+        }
+        |> Async.StartAsTask
+        
+    [<Benchmark>]
+    member x.ArrayRemoveManyAtBeginning() =
+        async {
+            Array.init x.Length (fun _ -> 0)
+            |> Array.removeManyAt 0 10 
+            |> ignore
+        }
+        |> Async.StartAsTask
+
+    [<Benchmark>]
+    member x.ArrayRemoveManyAtEnd() =
+        async {
+            Array.init x.Length (fun _ -> 0)
+            |> Array.removeManyAt (x.Length - 11) 10 
+            |> ignore
+        }
+        |> Async.StartAsTask
+        
+    [<Benchmark>]
+    member x.ArrayInsertAtBeginning() =
+        async {
+            Array.init x.Length (fun _ -> 0)
```

Comment:
It would be best if the benchmarks only measured the operation (Array.insertAt) and not things like Array.init.

Reply:
I changed all the methods to use an array/list which is pre-initialised.